CreeDictionary/API/search/core.py

```diff
@@ -37,7 +40,12 @@ def add_result(self, result: types.Result):
 
     def sorted_results(self) -> list[types.Result]:
         results = list(self._results.values())
-        results.sort(key=ranking.sort_by_user_query(self))
+
+        for r in self._results.values():
+            assign_relevance_score(r)
+        results.sort(key=attrgetter("relevance_score"))
+        results.reverse()
```

Comment:
Could (should?) this logic be moved into the `Result` class? That is, sorting a list of `Result` objects should automatically sort by relevance.

Reply:
Good idea! I hadn’t thought of that because I was keeping the new search method optional until very recently.